### PR TITLE
Update First App on a Mac 

### LINKED ISSUE #1842: tutorials/your-first-mac-aspnet.md "Your First ASP.NET Core Application on a Mac Using Visual Studio Code"
Add content to the Your First App on Mac tutorial to discuss the various artifacts in the generated project
Right now most of the content in the [Your First App on Mac tutorial](https://docs.asp.net/en/latest/tutorials/your-first-mac-aspnet.html) talks about Yeoman, Visual Studio Code and Azure. We should add content that actually goes through what is in the generate project.

### LINKED ISSUE #2726: csproj conversion aspnetcore/tutorials/your-first-mac-aspnet.md
[Your First ASP.NET Core Application on a Mac Using Visual Studio Code](https://docs.microsoft.com/en-us/aspnet/core/tutorials/your-first-mac-aspnet)
This upgrade requires a total rewrite.

-  Replace Yeoman with dotnet
- make it work with Linux

### PR DESCRIPTION
**Pull request by spboyer**
- Switch yeoman for dotnet cli
- update screen shots
- Fixes #2726 
- added information about files that are generated. Fixes #1842 

No code samples to update.

cc:/ @danroth27 @Rick-Anderson @tdykstra

### CODE REVIEW

File: aspnetcore/tutorials/your-first-mac-aspnet.md [hunk 1]
```diff
@@ -1,81 +1,71 @@
...
+## Setting up your development environment
 
-## Scaffolding Applications Using Yeoman
+To setup your development machine download and install [.NET Core](https://microsoft.com/net/core) and [Visual Studio Code](https://code.visualstudio.com) with the [C# extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp).
 
-We will be using `yo aspnet` to generate the **Web Application Basic** template, you may follow the full instructions in [Building Projects with Yeoman](../client-side/yeoman.md) to create an ASP.NET Core project which show an **Empty Web** for reference.
+## Scaffolding applications using dotnet new
 
-Install the necessary yeoman generators and bower using npm.
+We will be using `dotnet new` to generate a new web application using the "Empty Web Template". Create a working directory for your project called _firstmacapp_ and change the directory to the newly created folder.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: Create a working directory for your project called *firstmacapp* . `cd` to *firstmacapp*.
Or whatever it is on the MAC. Try to keep sentences short.

File: aspnetcore/tutorials/your-first-mac-aspnet.md [hunk 2]
```diff
@@ -1,81 +1,71 @@
...
 
-```console
-npm install -g yo generator-aspnet bower
-   ```
+Start Visual Studio Code and open the _firstmacapp_ folder. Press Ctrl + '`' (the back-quote character) to open an embedded terminal in VS Code. (Alternatively, you can use a separate terminal window, if you prefer).
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: Alternatively, use a separate terminal window.

File: aspnetcore/tutorials/your-first-mac-aspnet.md [hunk 3]
```diff
@@ -1,81 +1,71 @@
...
 
-Run the ASP.NET Core generator
+Run the `dotnet new` command to create a new web application, passing the `mvc` parameter as the template type.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: Run the following command:

`dotnet new mvc`

The `mvc` parameter creates a new MVC web app.
We don't like to tell them the command/code to use in a sentence then repeat it as code. The code is the easiest to understand - so use code then add sentences to expand on the code.

File: aspnetcore/tutorials/your-first-mac-aspnet.md [hunk 4]
```diff
@@ -1,81 +1,71 @@
...
 
-* Start **Visual Studio Code**
+* Startup.cs : [Startup Class](../fundamentals/startup.md) - class configures the request pipeline that handles all requests made to the application.
+* Program.cs : [Program Class](../fundamentals/) that contains the Main entry point of the application.
+* firstmacapp.csproj : [Project file](https://docs.microsoft.com/en-us/dotnet/articles/core/preview3/tools/csproj) MSBuild Project file format for ASP.NET Core applications. Contains Project to Project references, NuGet References and other project related items.
+* appsettings.json / appsettings.Development.json : Environment base app settings configuration file. [See Configuration](../fundamentals/configuration).
+* bower.json : Bower package dependencies for the project.
+* .bowerrc : Bower configuration file which defines where to install the components when Bower downloads the assets.
+* bundleconfig.json : configuration files for bundling and minifying front-end JavaScript and CSS assets.
+* Views : Contains the Razor views for the Controllers as well as Shared Views. Views are the components that display the app's user interface (UI). Generally, this UI displays the model data.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: **as well** banned
 Contains the Razor views for the controllers ~~as well as~~ and  shared views.
Why can't it be
 Contains the [Razor views](xref:mvc/views/razor)

File: aspnetcore/tutorials/your-first-mac-aspnet.md [hunk 5]
```diff
@@ -1,81 +1,71 @@
...
+* Controllers : Contains MVC Controllers, initially _HomeController.cs_. Controllers are Classes that handle browser requests, retrieve model data, and then specify view templates that return a response to the browser.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: classes lower case
I would shorten it to:
Controllers are [classes that handle browser requests](xref:mvc/controllers/actions).

File: aspnetcore/tutorials/your-first-mac-aspnet.md [hunk 6]
```diff
@@ -1,81 +1,71 @@
...
+* wwwroot : Web application root folder.
 
-* Tap **File > Open** and navigate to your ASP.NET Core app
+For more on the MVC pattern see [What is the MVC pattern?](../mvc/overview).
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: links need to use xref, not relative syntax
`[What is the MVC pattern?](xref:mvc/overview)`.

Comment (Rick-Anderson) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: For more information see `[What is the MVC pattern?](xref:mvc/overview)`.

File: aspnetcore/tutorials/your-first-mac-aspnet.md [hunk 7]
```diff
@@ -1,81 +1,71 @@
...
 
-![File menu](your-first-mac-aspnet/_static/file-open.png)
+## Developing ASP.NET Core applications on a Mac with Visual Studio Code
 
-When the application is opened, Visual Studio Code will prompt to restore the needed project dependencies as well as add build and debug dependencies.
+Select the _Program.cs_ file and Visual Studio Code will prompt to restore the needed project dependencies as well as add build and debug dependencies.
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: Don't we use `*` for italic?

Comment (Rick-Anderson) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: Add **as well as** to the banned list

Select *Program.cs* and VS Code will prompt to restore the needed project dependencies **and** add build and debug dependencies.

Comment (Rick-Anderson) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: @tdykstra  is it OK to  on first use call it Visual Studio Code (VS Code) and after that use the industry jargon VS Code?

File: aspnetcore/tutorials/your-first-mac-aspnet.md (hunk 3 above)
Comment (spboyer) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: I think that the change is a robotic. It is important to note that `mvc` is a parameter, and give the reader the information or lead them that there are other types. Not to give the impression that `dotnet new mvc` is the only way to create an application.

Thoughts?

File: aspnetcore/tutorials/your-first-mac-aspnet.md (hunk 5 above)
Comment (spboyer) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: Made the change, FYI -  copied that text from /mvc/overview.md

File: aspnetcore/tutorials/your-first-mac-aspnet.md (hunk 7 above)
Comment (spboyer) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: changed but both syntax is supported:  _underscore_ *single-asterix*

File: aspnetcore/tutorials/your-first-mac-aspnet.md (hunk 4 above)
Comment (spboyer) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: changed

File: aspnetcore/tutorials/your-first-mac-aspnet.md (hunk 2 above)
Comment (spboyer) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: changed

File: aspnetcore/tutorials/your-first-mac-aspnet.md (hunk 1 above)
Comment (spboyer) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: done.

File: aspnetcore/tutorials/your-first-mac-aspnet.md (hunk 7 above)
Comment (tdykstra) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: @spboyer VS Code Markdown editor uses underscores for italics - is there a way to make it use asterisks?

File: aspnetcore/tutorials/your-first-mac-aspnet.md (hunk 6 above)
Comment (tdykstra) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: I didn't think we had a consensus on making xref format required everywhere.

File: aspnetcore/tutorials/your-first-mac-aspnet.md (hunk 3 above)
Comment (tdykstra) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: < It is important to note that  mvc  is a parameter, and give the reader the information or lead them that there are other types. Not to give the impression that  dotnet new mvc  is the only way to create an application.

I agree, I think the proposed text isn't clear enough.

Comment (tdykstra) [inline aspnetcore/tutorials/your-first-mac-aspnet.md]: > It is important to note that  mvc  is a parameter, and give the reader the information or lead them that there are other types. Not to give the impression that  dotnet new mvc  is the only way to create an application.

I agree, I think the proposed text isn't clear enough.

**Pull request merged**